### PR TITLE
jerry: support binary literal

### PR DESCRIPTION
**Pull request by ErosZy**
[#446](https://github.com/Rokid/ShadowNode/issues/446)

I dont find jerry/number test file, where should i add ?

### CODE REVIEW

**Comment by ErosZy** [comment]
Oh sorry, i forget reset my branch :(, need to be create a new pr?

**Comment by yorkie** [comment]
Just need a rebase, it's up to you @ErosZy :p

**Comment by ErosZy** [comment]
> Looks like you picked changes from master branch to your PR branch. So these changes are also included in your PR.
> 
> A rebase could be performed to remove unnecessary commits from your PR branch.

done, thanks @legendecas

**Comment by yorkie** [comment]
We need tests for ECMAScript at https://github.com/Rokid/ShadowNode/tree/master/deps/jerry/tests/jerry :)

**Comment by legendecas** [comment]
> I dont find jerry/number test file, where should i add ?

You could add tests file to `deps/jerry/tests/jerry`. Then run `deps/jerry/tools/run-tests.py --jerry-tests` to run all jerry test cases.

**Comment by ErosZy** [comment]
> > I dont find jerry/number test file, where should i add ?
> 
> You could add tests file to `deps/jerry/tests/jerry`. Then run `deps/jerry/tools/run-tests.py --jerry-tests` to run all jerry test cases.

Ok, I just thought need to create a c file to test. i will add test for this. @yorkie @legendecas

**Comment by ErosZy** [comment]
@legendecas @yorkie test file commit, and `npm test` pass in my local :)

File: deps/jerry/jerry-core/parser/js/js-lexer.c [hunk 1]
```diff
@@ -1756,7 +1781,25 @@ lexer_construct_number_object (parser_context_t *context_p, /**< context */
   uint32_t literal_index = 0;
   prop_length_t length = context_p->token.lit_location.length;
 
-  if (context_p->token.extra_value != LEXER_NUMBER_OCTAL)
+  if(context_p->token.extra_value == LEXER_NUMBER_BINARY)
```
Comment (yorkie) [inline deps/jerry/jerry-core/parser/js/js-lexer.c]:
```suggestion
  if (context_p->token.extra_value == LEXER_NUMBER_BINARY)
```

File: deps/jerry/jerry-core/parser/js/js-lexer.c [hunk 2]
```diff
@@ -1756,7 +1781,25 @@ lexer_construct_number_object (parser_context_t *context_p, /**< context */
...
+  {
+    const uint8_t *src_p = context_p->token.lit_location.char_p;
+    const uint8_t *src_end_p = src_p + length - 1;
+
+    if(LEXER_TO_ASCII_LOWERCASE (src_p[1]) == LIT_CHAR_LOWERCASE_B)
```
Comment (yorkie) [inline deps/jerry/jerry-core/parser/js/js-lexer.c]:
```suggestion
    if (LEXER_TO_ASCII_LOWERCASE (src_p[1]) == LIT_CHAR_LOWERCASE_B)
```

File: deps/jerry/jerry-core/parser/js/js-lexer.c [hunk 3]
```diff
@@ -1756,7 +1781,25 @@ lexer_construct_number_object (parser_context_t *context_p, /**< context */
...
+      src_p++;
+    }
+
+    num = 0;
+    do
+    {
+      src_p++;
+      num = num * 2 + (ecma_number_t) (*src_p - LIT_CHAR_0);
+    }
+    while (src_p < src_end_p);
```
Comment (yorkie) [inline deps/jerry/jerry-core/parser/js/js-lexer.c]: @ErosZy See https://github.com/Rokid/ShadowNode/blob/master/deps/jerry/jerry-core/parser/js/js-lexer.c#L1766-L1781, I think we should clean up the `if-else` cases.

**Comment by ErosZy** [comment]
all done, thanks @yorkie

**Comment by legendecas** [comment]
Hi, would you like to merge changes on master branch to enable jerry tests on CI? :P jerry tests are enabled on https://github.com/Rokid/ShadowNode/pull/470.

**Comment by ErosZy** [comment]
> Hi, would you like to merge changes on master branch to enable jerry tests on CI? :P jerry tests are enabled on #470.

No problem.

**Comment by ErosZy** [comment]
@legendecas merge jerry/test done, and pass my workspace :)

**Pull request merged**